conda/_vendor/auxlib/type_coercion.py

```diff
@@ -255,6 +255,9 @@ def typify_data_structure(value, type_hint=None):
         return type(value)((k, typify(v, type_hint)) for k, v in iteritems(value))
     elif isiterable(value):
         return type(value)(typify(v, type_hint) for v in value)
+    elif (isinstance(value, string_types)
+          and isinstance(type_hint, type) and issubclass(type_hint, string_types)):
+        return type_hint(value)
```

Comment:
This had to be added because we were calling `.strip()` otherwise.  For now, I think this is the least invasive place to make a change.  It's very likely that we if remove `.strip()` from `typify()`, we'll destabilize something somewhere else that's relying on that behavior.

Reply:
Can you add a comment to the code describing why we need that? It seems like something we might possibly revise at some point -- a comment would be helpful to remind us of the "trailing space in prompt modifier" use case.